src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/TypedReferenceHelpers.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Internal.Runtime.CompilerHelpers
+{
+    /// <summary>
+    /// These methods are used to implement TypedReference-related instructions.
+    /// </summary>
+    internal static class TypedReferenceHelpers
+    {
+        public unsafe static RuntimeTypeHandle TypeHandleToRuntimeTypeMaybeNull(RuntimeTypeHandle typeHandle)
```

Comment:
It's what we give to RyuJIT as `CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPE_MAYBENULL`.